### PR TITLE
xcompose: add module

### PR DESCRIPTION
**Pull request by dawidsowa**
This module allows generating .XCompose file

### CODE REVIEW

File: modules/services/xcompose.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  xComposeType = with types; either (attrsOf xComposeType) str;
```
Comment (rycee) [inline modules/services/xcompose.nix]: You can't have recursive types. Perhaps once `lazyAttrsOf` is available?

Comment (dawidsowa) [reply modules/services/xcompose.nix]: Ok, I'll wait then

Comment (bb010g) [reply modules/services/xcompose.nix]: `lazyAttrsOf` has been merged in NixOS/nixpkgs#70138.

Comment (dawidsowa) [reply modules/services/xcompose.nix]: Thanks for the info

**Pull request closed without merging**